### PR TITLE
Added Create New Password Field to Import Screens

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
Added Create New Password Fields to the Onboarding Import Screens for importing from a `MetaMask` or `BraveLegacyWallet`

1) Now Requires to enter a new `Password` on the import screen
2) Gives the option to use same `Password` of the importing wallet, but if it does not meet the new wallet password requirements, will force the user to create a new password that does.

Resolves <https://github.com/brave/brave-browser/issues/18264>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

https://user-images.githubusercontent.com/40611140/134741210-c2d83d11-f17f-45fb-ad17-7c57c68a8b35.mov

## Test Plan:
Please reference the test plan in this pull request https://github.com/brave/brave-core/pull/10161
but with some additional steps for `MetaMask` and `Crypto Wallet`

New Wallet Password Requirements are
`Passwords must be at least 7 characters containing at least one number and a special character.`

### Additional Steps:
1. Input different password from ("referenced  step 1") and input a new password meeting password requirements, click `Import`.
2. There will be an import error message.
3. Input a password that does not meet password requirements into the import password field and click the `Use the same password` check box.
4. The UI should change from `Set a Brave Wallet password` too `A more secure password is required for Brave Wallet` and require you to create a new password that meets the requirements.
5. Input a password that meets password requirements into the import password field and click the `Use the same password` check box.
6. UI will hide the new password fields and allow you to continue or you can uncheck `Use the same password` and enter a new password.
7. Input correct password from ("referenced  step 1"), if you know that that your password meets requirements and want to use it, click the `Use the same password` check box or create a new password.
8. Click `Import`.
8. The wallet will be setup with your new or same password, depending on your preference.
9. Check the mnemonic is same as ("referenced  step 1")

### CODE REVIEW

File: components/brave_wallet_ui/stories/wallet-concept.tsx [hunk 1]
```diff
@@ -201,11 +201,11 @@ const transactionDummyData: TransactionListInfo[] = [
 
 export const _DesktopWalletConcept = (args: { onboarding: boolean, locked: boolean }) => {
   const {
-    onboarding,
+    // onboarding,
     locked
   } = args
   const [view] = React.useState<NavTypes>('crypto')
-  const [needsOnboarding, setNeedsOnboarding] = React.useState<boolean>(onboarding)
+  const [needsOnboarding, setNeedsOnboarding] = React.useState<boolean>(true)
```
Comment (onyb) [inline components/brave_wallet_ui/stories/wallet-concept.tsx:208]: Is this change intended?

File: components/brave_wallet_ui/stories/screens/onboarding.tsx [hunk 2]
```diff
@@ -160,11 +190,20 @@ function Onboarding (props: Props) {
         {isImporting &&
           <OnboardingImportMetaMaskOrLegacy
             onSubmit={onImport}
+            onImportPasswordChanged={handleImportPasswordChanged}
             onPasswordChanged={handlePasswordChanged}
-            disabled={password === ''}
+            onConfirmPasswordChanged={handleConfirmPasswordChanged}
+            disabled={checkConfirmedPassword || checkPassword || password === '' || confirmedPassword === '' || importPassword === ''}
```
Comment (onyb) [inline components/brave_wallet_ui/stories/screens/onboarding.tsx:196]: Let's have a separate variable for this.

**Review by onyb — APPROVED** [review]
Pre-approving, but one small comment to address.

File: components/brave_wallet_ui/stories/screens/onboarding.tsx (hunk 2 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/stories/screens/onboarding.tsx]: Done!

File: components/brave_wallet_ui/stories/wallet-concept.tsx (hunk 1 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/stories/wallet-concept.tsx]: Fixed!

File: components/brave_wallet_ui/constants/locale.ts [hunk 3]
```diff
@@ -295,6 +295,11 @@ const locale = {
   connectWithSiteNext: 'Next',
   connectWithSiteHeaderTitle: 'Connect With Brave Wallet',
 
+  // Import from Hardware
```
Comment (darkdh) [inline components/brave_wallet_ui/constants/locale.ts:298]: it is not from hardware though, let's just say external wallets

Comment (Douglashdaniel) [reply components/brave_wallet_ui/constants/locale.ts]: Updated!

**Comment by darkdh** [comment]
shouldn't we hide the new password and re-enter input fields when users check "Use same password" and password for importer meets the requirement?

**Comment by Douglashdaniel** [comment]
> shouldn't we hide the new password and re-enter input fields when users check "Use same password" and password for importer meets the requirement?

Updated! 👍

File: components/brave_wallet_ui/components/desktop/wallet-onboarding/import-meta-mask-or-legacy/index.tsx [hunk 4]
```diff
@@ -9,29 +9,54 @@ import {
...
 export interface Props {
   onSubmit: () => void
   onPasswordChanged: (value: string) => void
+  onConfirmPasswordChanged: (value: string) => void
+  onImportPasswordChanged: (value: string) => void
   onClickLost: () => void
+  onCheckedPassword: (checked: boolean) => void
+  password: string
+  confirmedPassword: string
+  checkedPassword: boolean
```
Comment (darkdh) [inline components/brave_wallet_ui/components/desktop/wallet-onboarding/import-meta-mask-or-legacy/index.tsx:30]: this naming is confusing, I thought it was checking if it meets password requirement when I first saw it.
It's better to rename it as `useSamePassword`

File: components/brave_wallet_ui/page/container.tsx [hunk 5]
```diff
@@ -547,14 +547,14 @@ function Container (props: Props) {
     props.walletPageActions.doneViewingPrivateKey()
   }
 
-  const onImportCryptoWallets = (password: string) => {
+  const onImportCryptoWallets = (password: string, newPassword: string) => {
     // TODO(Douglashdaniel): Use different set of password for new password
```
Comment (darkdh) [inline components/brave_wallet_ui/page/container.tsx:551]: this TODO can be removed

File: components/brave_wallet_ui/constants/locale.ts [hunk 6]
```diff
@@ -109,10 +109,10 @@ const locale = {
 
   // Create Password
   createPasswordTitle: 'Secure your crypto with a password',
-  createPasswordDescription: 'Password must be at least 7 characters containing at least one number and a special character',
+  createPasswordDescription: 'Passwords must be at least 7 characters containing at least one number and a special character.',
   createPasswordInput: 'Password',
   createPasswordInput2: 'Re-Type Password',
```
Comment (darkdh) [inline components/brave_wallet_ui/constants/locale.ts:114]: this should be `confirmPasswordInput`

File: components/brave_wallet_ui/constants/locale.ts [hunk 7]
```diff
@@ -109,10 +109,10 @@ const locale = {
...
-  createPasswordError: 'Password must be 7 or more characters',
+  createPasswordError: 'Passwords must be at least 7 characters containing at least one number and a special character.',
   createPasswordError2: 'Re-typed password does not match',
```
Comment (darkdh) [inline components/brave_wallet_ui/constants/locale.ts:116]: `confirmPasswordError`

File: components/brave_wallet_ui/page/container.tsx [hunk 8]
```diff
@@ -547,14 +547,14 @@ function Container (props: Props) {
...
-    props.walletPageActions.importFromCryptoWallets({ password, newPassword: password })
+    props.walletPageActions.importFromCryptoWallets({ password, newPassword })
   }
 
-  const onImportMetaMask = (password: string) => {
+  const onImportMetaMask = (password: string, newPassword: string) => {
     // TODO(Douglashdaniel): Use different set of password for new password
```
Comment (darkdh) [inline components/brave_wallet_ui/page/container.tsx:556]: this TODO can be removed

File: components/brave_wallet_ui/components/desktop/wallet-onboarding/import-meta-mask-or-legacy/index.tsx (hunk 4 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/components/desktop/wallet-onboarding/import-meta-mask-or-legacy/index.tsx]: Changed!

File: components/brave_wallet_ui/page/container.tsx (hunk 5 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/container.tsx]: Removed!

File: components/brave_wallet_ui/constants/locale.ts (hunk 6 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/constants/locale.ts]: Changed!

File: components/brave_wallet_ui/constants/locale.ts (hunk 7 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/constants/locale.ts]: Changed!

File: components/brave_wallet_ui/page/container.tsx (hunk 8 above)
Comment (Douglashdaniel) [reply components/brave_wallet_ui/page/container.tsx]: Removed!

**Review by darkdh — APPROVED** [review]

**Review by bbondy — APPROVED** [review]
